Plugins/Rename/NWScript/nwnx_rename.nss

```diff
@@ -2,18 +2,29 @@
 
 const string NWNX_Rename = "NWNX_Rename";
 
-//Set a player's floaty name(sPrefix+sNewName+sSuffix) and name on the player list (sNewName). Will not persist through saving, resets or logout.
-void NWNX_Rename_SetPlayerNameOverride(object oPlayer, string sPrefix, string sNewName, string sSuffix);
+//Constants for Player Name states.
+const int NWNX_RENAME_PLAYERNAME_DEFAULT = 0;
+const int NWNX_RENAME_PLAYERNAME_OBFUSCATE = 1;
+const int NWNX_RENAME_PLAYERNAME_OVERRIDE = 2;
 
+//Set a PC's floaty/chat name(sPrefix+sNewName+sSuffix) and name (sNewName) on the player list.
+//If (iPlayerNameState) is set to NWNX_RENAME_PLAYERNAME_OVERRIDE the player name will change to (sNewName) on tells.
+//If (iPlayerNameState) is set to NWNX_RENAME_PLAYERNAME_OBFUSCATE the player name will be set to a random string.
+//If (iPlayerNameState) is set to NWNX_RENAME_PLAYERNAME_DEFAULT the player name will be untouched.
+//DMs and the player themselves will still see their original player name regardless of option set
+//Will not persist through saving, resets or logout.
+void NWNX_Rename_SetPCNameOverride(object oPC, string sNewName, string sPrefix = "" , string sSuffix = "" , int iPlayerNameState = NWNX_RENAME_PLAYERNAME_DEFAULT);
 
-void NWNX_Rename_SetPlayerNameOverride(object oPlayer, string sPrefix, string sNewName, string sSuffix)
+
+void NWNX_Rename_SetPCNameOverride(object oPC, string sNewName, string sPrefix = "" , string sSuffix = "" , int iPlayerNameState = NWNX_RENAME_PLAYERNAME_DEFAULT)
 {
-    string sFunc = "SetPlayerNameOverride";
+    string sFunc = "SetPCNameOverride";
     
+    NWNX_PushArgumentInt(NWNX_Rename, sFunc, bOverridePlayerName);
```

Comment:
Did you mean `iPlayerNameState` instead of `bOverridePlayerName` here?

Reply:
I indeed mean that.